File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    stack = []

    if s[0] == ')':
        return False
    
    else:
        for i in range(len(s)):
            if s[i] == '(': # push
                stack.append(s[i])
        
            elif not len(stack): # empty check
                return False
            
            else: # pop
                stack.pop(-1)
        
        answer = True if not len(stack) else False
        return answer

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.cpp
#include <string>
#include <stack>
#include <iostream>

using namespace std;

bool solution(string s)
{   
    if(s[0] == ')') return false; // 바로 리턴
    
    bool answer = true;
    stack<char> check;
    
    for(char c: s){
        if(c == '('){
            check.push(c);
        }
        
        else{
            if(!check.empty() && c == ')' && check.top() == '('){
                check.pop();
            }
        }
        
    }
    
    if(check.empty()){
        return true;
    } else{
        return false;
    }
}